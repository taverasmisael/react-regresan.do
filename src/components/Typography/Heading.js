import React from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'

const Heading = ({ children, color = 'inherit', truncate = false, align = 'left', size = '1' }) => (
  <Typography gutterBottom type={`display${size}`} color={color} align={align} noWrap={truncate}>
    {children}
  </Typography>
)

Heading.propTypes = {
  children: PropTypes.string,
  align: PropTypes.oneOf(['inherit', 'left', 'center', 'right', 'justify']),
  color: PropTypes.oneOf(['inherit', 'primary', 'secondary', 'accent', 'error', 'default']),
  size: PropTypes.oneOf(['1', '2', '3', '4']),
  truncate: PropTypes.bool
}

export default Heading
