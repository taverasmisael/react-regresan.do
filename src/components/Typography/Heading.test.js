import React from 'react'
import { shallow } from 'enzyme'

import Heading from './Heading'

describe('components/Typography/Heading', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<Heading />)
    })
    it('matches snapshot', () => {
      expect(<Heading />).toMatchSnapshot()
      expect(<Heading>Hello World</Heading>).toMatchSnapshot()
    })
  })

  describe('variants', () => {
    it('text display correctly', () => {
      let component = shallow(<Heading size="2">Gangsta</Heading>)
      expect(component.props().type).toBe('display2')
      component = shallow(<Heading size="3">Gangsta</Heading>)
      expect(component.props().type).toBe('display3')
    })
  })
})
