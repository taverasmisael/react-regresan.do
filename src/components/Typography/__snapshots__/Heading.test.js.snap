// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/Heading rendering matches snapshot 1`] = `<Heading />`;

exports[`components/Typography/Heading rendering matches snapshot 2`] = `
<Heading>
  Hello World
</Heading>
`;
